src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,99 @@
+package duke.util;
+
+import duke.exception.DukeException;
+
+import java.util.Scanner;
+
+public class Parser {
+    private final Scanner sc;
+    private final List list;
+
+    public Parser(List list) {
+        this.sc = new Scanner(System.in);
+        this.list = list;
+    }
+
+    // extract int from done and delete prompts
```

Comment:
I believe public non-trivial methods such as this (as well as the others in this class and other classes) should be well-documented by Javadocs. Do consider adding them!